### PR TITLE
Add Gallery layout

### LINKED ISSUE #74: Create a TileBoard/Gallery layout
A TileBoard layout is a layout where all of the children are a uniform size, the children are displayed horizontally, and they wrap as needed. It's really just a grid, but the name is already taken.

Example:

image

### PR DESCRIPTION
**Pull request by cliffpyles**
A Gallery layout is a layout where all of the children are a uniform size, the children are displayed horizontally, and they wrap as needed. It's really just a grid, but the name is already taken.

resolve #74

### CODE REVIEW

File: src/patternfly/layouts/Gallery/styles.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.pf-l-gallery {
+  display: grid;
+  grid-gap: 1rem; //revisit this value
```
Comment (andresgalante) [inline src/patternfly/layouts/Gallery/styles.scss]: This should be a variable, something like "gutter" to use on different layouts and it should be themable.

File: src/patternfly/layouts/Gallery/styles.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
```
Comment (andresgalante) [inline src/patternfly/layouts/Gallery/styles.scss]: 250px is randome but ok for now

Comment (cliffpyles) [reply src/patternfly/layouts/Gallery/styles.scss]: Yeah. I meant to leave a comment by that line also.

File: src/patternfly/layouts/Gallery/styles.scss (hunk 1 above)
Comment (cliffpyles) [reply src/patternfly/layouts/Gallery/styles.scss]: Correct, but we originally said that layouts wouldn't pull in global variables.

Comment (cliffpyles) [reply src/patternfly/layouts/Gallery/styles.scss]: Do you mean you just want a the component variable for now?

Comment (andresgalante) [reply src/patternfly/layouts/Gallery/styles.scss]: yeap, we did say that. But we need a variable anyway. maybe this is an exception

Comment (andresgalante) [reply src/patternfly/layouts/Gallery/styles.scss]: yes, a component type variable that calls a global variable

Comment (cliffpyles) [reply src/patternfly/layouts/Gallery/styles.scss]: If I remove the grid-gap all together, we can shift responsibility to the padding of the child, and then an exception wouldn't be necessary.

If we take that route, we could also circle back later after we've had time to think through introduction of global variables and how that will affect the system as a whole.

What do you think?

**Comment by andresgalante** [comment]
I'll merge it for now. lets talk about the gap variable next week :)

**Pull request merged**